### DISCUSSION TITLE
Is there a way to use pay-rails with GooglePay and ApplePay?

### DISCUSSION THREAD

**Discussion by vuniverseklaus** [pay-rails/pay#595] [Q&A]
I'm following this doc https://stripe.com/docs/google-pay for react-native
It requires a PaymentIntent, I'm doing it by setting the value of the **subscription pricing**, all working fine but some things does not happening:

Does not relate the paid intent to the logged user (this doc does not explain how to relate a payment with an api logged user).
Does not set a subscription for the user as paid.

I have all flow working with web integration using pay-rails using this piece of code:
```
...
if @current_user.payment_processor.subscribed?
      if @current_user.payment_processor.subscription.swap(price_id)
      ...
else
  @current_user.payment_processor.checkout(
          mode: "subscription",
...
```

This method returns an url to redirect to stripe but using Google Pay will not use a redirect.
I think pay-rails checkout/webhook do all relation between user and customer/subscription
so I don't know if I can use some method from pay-rails that handles creating the paymentIntent and receiving the hook from Stripe GooglePay paid...

Someone knows if can I use some method for this? or I can't use pay-rails with Google Pay / Apple Pay

**Comment by excid3** [comment]
Google Pay and Apple Pay are handled client side in the Javascript or in native app code, not on the server where Pay is used.

**Reply by vuniverseklaus** [reply]
Yes, I know that, but I think pay-rails could have some method like payment intent that sends metadata with user_id, subscription_id and handle it on pay-rails hook by creating the customer/subscription when Stripe send to the webhook that payment intent is paid...
I am thinking about creating Customer manually in Pay models:

```
Pay::Customer:0x00007f8d2356c158
  id: 1,
  owner_type: "User",
  owner_id: 1,
  processor: "stripe",
  processor_id: "cus_...",
```
and

```
Pay::Subscription:0x00007f8d275798b8
 id: 1,
 customer_id: 1,
 name: "default",
 processor_id: "sub_..",
 processor_plan: "price_...",
 quantity: 1,
```

otherwise I should verify if user has subscribed by web (using pay-rails) or mobile Google Play Apple Pay, to handle or not using user.**payment_processor**

thanks for your answer
